src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -134,68 +141,33 @@ private function extractFromMutator($class, $property)
         $reflectionParameters = $reflectionMethod->getParameters();
         $reflectionParameter = $reflectionParameters[0];
 
-        $arrayMutator = in_array($prefix, self::$arrayMutatorPrefixes);
-
-        if (method_exists($reflectionParameter, 'getType') && $reflectionType = $reflectionParameter->getType()) {
-            $fromReflectionType = $this->extractFromReflectionType($reflectionType);
-
-            if (!$arrayMutator) {
-                return array($fromReflectionType);
+        if ($this->supportsParameterType) {
+            if (!$reflectionType = $reflectionParameter->getType()) {
+                return;
             }
+            $type = $this->extractFromReflectionType($reflectionType);
 
-            $phpType = Type::BUILTIN_TYPE_ARRAY;
-            $collectionKeyType = new Type(Type::BUILTIN_TYPE_INT);
-            $collectionValueType = $fromReflectionType;
-        }
-
-        if ($reflectionParameter->isArray()) {
-            $phpType = Type::BUILTIN_TYPE_ARRAY;
-            $collection = true;
-        }
-
-        if ($arrayMutator) {
-            $collection = true;
-            $nullable = false;
-            $collectionNullable = $reflectionParameter->allowsNull();
-        } else {
-            $nullable = $reflectionParameter->allowsNull();
-            $collectionNullable = false;
-        }
-
-        if (!isset($collection)) {
-            $collection = false;
-        }
-
-        if (method_exists($reflectionParameter, 'isCallable') && $reflectionParameter->isCallable()) {
-            $phpType = Type::BUILTIN_TYPE_CALLABLE;
-        }
-
-        if ($typeHint = $reflectionParameter->getClass()) {
-            if ($collection) {
-                $phpType = Type::BUILTIN_TYPE_ARRAY;
-                $collectionKeyType = new Type(Type::BUILTIN_TYPE_INT);
-                $collectionValueType = new Type(Type::BUILTIN_TYPE_OBJECT, $collectionNullable, $typeHint->name);
+            // HHVM reports variadics with "array" but not builtin type hints
+            if (!$reflectionType->isBuiltin() && Type::BUILTIN_TYPE_ARRAY === $type->getBuiltinType()) {
+                return;
+            }
+        } elseif (preg_match('/^(?:[^ ]++ ){4}([a-zA-Z_\x7F-\xFF][^ ]++)/', $reflectionParameter, $info)) {
```

Comment:
By using the string representation of the parameter to get the type hint (on PHP5 only of course), we prevent autoloading of the corresponding class.